chore(feature flags docs): clarify "Persist flag" option (#9210)
PostHog/posthog.com: contents/docs/feature-flags/creating-feature-flags.mdx

### Persisting feature flags across authentication steps

> > **Note:** This option introduces trade-offs that are not worthwhile for the majority of our users.

Reviewer: Since we say this later, we can remove from here
```suggestion 
```

> **Note:** This is only relevant if your feature flag is shown to both logged out AND logged in users.

Feature flag values are calculated based on a user's properties. Since it's possible for a user to have different properties before and after login, they may receive different feature flag values before and after logging in.

By enabling the option to persist feature flags across authentication, you ensure that the flag value remains the same. 

> There are trade-offs to enabling this. In our experience, these tradeoffs are not worthwhile for the majority of our users:

Reviewer: ```suggestion
In our experience, the tradeoffs to enabling this are **not** worthwhile for the majority of our users. They include:
```

1. **Slower feature flag response:** Enabling this option introduces additional checks, which can slow down the response time when fetching feature flags.

> 2. **Incompatible with [Local evaluation](/docs/feature-flags/local-evaluation):** It is necessary to perform the additional checks on the PostHog servers.

Reviewer: ```suggestion
2. **Incompatible with [local evaluation](/docs/feature-flags/local-evaluation):** It is necessary to perform the additional checks on the PostHog servers.
```

> 3. **Incompatible with [Bootstrapping](/docs/feature-flags/bootstrapping):** Bootstrapping relies on local evaluation, which cannot correctly calculate persistent flag values.

Reviewer: ```suggestion
3. **Incompatible with [bootstrapping](/docs/feature-flags/bootstrapping):** Bootstrapping relies on local evaluation, which cannot correctly calculate persistent flag values.
```
